targets/UnrealMarketplacePlugin/source/PlayFab/Source/PlayFabCommon/Public/PlayFabAuthenticationContext.h.ejs

```diff
@@ -0,0 +1,118 @@
+﻿<%- copyright %>
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "PlayFabAuthenticationContext.generated.h"
+
+/**
+* Container for PlayFab authentication credentials data.
+*/
+UCLASS(BlueprintType)
+class PLAYFABCOMMON_API UPlayFabAuthenticationContext : public UObject
+{
+    GENERATED_BODY()
+public:
+
+
+    // Get the client session ticket that is used as an authentication token in many PlayFab API methods.
+    UFUNCTION(BlueprintCallable, BlueprintPure, Category = "PlayFab | Core")
+        FString& GetClientSessionTicket()
+    {
+#if defined(DISABLE_PLAYFABCLIENT_API)
+#if DISABLE_PLAYFABCLIENT_API
+        checkf(false, TEXT("ClientSessionTicket is disabled because DISABLE_PLAYFABCLIENT_API is set."));
+#endif
+#endif
+        return ClientSessionTicket;
+    }
+
+    // Set the client session ticket that is used as an authentication token in many PlayFab API methods.
+    UFUNCTION(BlueprintCallable, Category = "PlayFab | Core")
+        void SetClientSessionTicket(FString InTicket)
+    {
+#if defined(DISABLE_PLAYFABCLIENT_API)
+#if DISABLE_PLAYFABCLIENT_API
+        checkf(false, TEXT("ClientSessionTicket is disabled because DISABLE_PLAYFABCLIENT_API is set."));
+#endif
+#endif
+        ClientSessionTicket = InTicket;
+    }
+
+
+    // Get the user's entity token. Entity tokens are required by all Entity API methods.
+    UFUNCTION(BlueprintCallable, BlueprintPure, Category = "PlayFab | Core")
+        FString& GetEntityToken()
+    {
+#if defined(DISABLE_PLAYFABENTITY_API)
+#if DISABLE_PLAYFABENTITY_API
+        checkf(false, TEXT("EntityToken is disabled because DISABLE_PLAYFABENTITY_API is set."));
+#endif
+#endif
+        return EntityToken;
+    }
+
+    // Set the user's entity token. Entity tokens are required by all Entity API methods.
+    UFUNCTION(BlueprintCallable, Category = "PlayFab | Core")
+        void SetEntityToken(FString InToken)
+    {
+#if defined(DISABLE_PLAYFABENTITY_API)
+#if DISABLE_PLAYFABENTITY_API
+        checkf(false, TEXT("EntityToken is disabled because DISABLE_PLAYFABENTITY_API is set."));
+#endif
+#endif
```

Comment:
Please remove all these double-#if's
I just want a single condition to test if the define exists, I never want to check the value of it.